test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,292 @@
+            client.askDialog
+          end
+
+          context "and value is a boolean" do
+            let(:value) { true }
+            let(:ask) { base_ask.merge("type" => "boolean", "file" => file_path)}
+
+            it "save the value in the file" do
+              expect(Yast::SCR).to receive(:Write).
+                with(Yast::Path.new(".target.string"), file_path, "true").
+                and_return(true)
+              client.askDialog
+            end
+          end
+        end
+
+        context "and a script was specified" do
+          let(:script) { { "source" => "echo", "filename" => "test.sh" } }
+          let(:ask) { base_ask.merge("script" => script) }
+          let(:tmp_dir) { "/tmp" }
+          let(:log_dir) { "/var/log/YaST2" }
+
+          before(:each) do
+            allow(Yast::AutoinstConfig).to receive(:tmpDir).and_return(tmp_dir)
+            allow(Yast::AutoinstConfig).to receive(:logs_dir).
+              and_return(log_dir)
+            script_path = "#{tmp_dir}/#{script["filename"]}"
+            allow(Yast::SCR).to receive(:Write).
+              with(Yast::Path.new(".target.string"), script_path, script["source"]).
+              and_return(true)
+            allow(Yast::SCR).to receive(:Execute).
+              with(Yast::Path.new(".target.mkdir"), File.join(tmp_dir, "ask_scripts_log"))
+          end
+
+          context "when environment is not set" do
+            it "runs the script without passing the ask value" do
+              expect(Yast::SCR).to receive(:Execute).
+                with(Yast::Path.new(".target.bash"),
+                     "/bin/sh -x /tmp/test.sh 2&> /tmp/ask_scripts_log/test.sh.log ")
+              client.askDialog
+            end
+          end
+
+          context "when environment is set" do
+            let(:script) { { "source" => "echo", "filename" => "test.sh", "environment" => true } }
+
+            it "runs the script passing the ask value" do
+              expect(Yast::SCR).to receive(:Execute).
+                with(Yast::Path.new(".target.bash"),
+                     "VAL=\"some-value\" /bin/sh -x /tmp/test.sh 2&> /tmp/ask_scripts_log/test.sh.log ")
+              client.askDialog
+            end
+          end
+
+          it "shows some feedback while script is running" do
+            message = "A user defined script is running. This may take a while."
+            allow(Yast::SCR).to receive(:Execute)
+            expect(Yast::Popup).to receive(:Feedback).
+              with("", client._(message))
+            client.askDialog
+          end
+        end
+
+        context "and more dialogs left" do
+          let(:ask_list) { [base_ask, base_ask.merge("dialog" => "1")] }
+
+          it "next dialog is shown" do
+            expect(Yast::UI).to receive(:UserInput).twice
+            client.askDialog
+          end
+        end
+
+        context "and no more dialogs left" do
+          let(:ask_list) { [base_ask] }
+
+          it "terminates" do
```

Comment:
what means terminates? It is quite strong word for me how to exit. Maybe simple it returns from method?